pkg/maintainer/maintainer.go

```diff
@@ -17,7 +17,13 @@ func Initialize(
 	launchAll := !config.BitcoinDifficulty
 
 	if config.BitcoinDifficulty || launchAll {
-		newBitcoinDifficultyMaintainer(ctx, btcChain, chain)
+		initializeBitcoinDifficultyMaintainer(
+			ctx,
+			btcChain,
+			chain,
```

Comment:
Non-blocking for this PR but we should think about the way of passing multiple host chain interfaces to `Initialize` function. `Initialize` function runs all or specific maintainers. Each maintainer will probably have its own host chain interface, e.g: the difficulty maintainer has `BitcoinDifficultyChain`, the wallet maintainer will have `WalletChain`, and so on.

We can approach that problem twofold:

**Option 1** 
We can introduce the following interface in the `chain.go` file
```
package maintainer

type Chain interface {
    BitcoinDifficultyChain
    WalletChain
}
```
Then, we can pass it to the `Initialize` function.

**Option 2**
We pass specific maintainer chain interfaces to `Initialize` function:
```
func Initialize(
	ctx context.Context,
	config Config,
	btcChain bitcoin.Chain,
	difficultyChain BitcoinDifficultyChain,
        walletChain WalletChain,
)
```

IMO, option 1 seems better at the moment. However, we don't have to decide right now and think about it during the development of the wallet maintainer (https://github.com/keep-network/keep-core/pull/3446).

Reply:
I agree, option 1 seems to be better. I'll implement it in #3446.

Reply:
Leaving it open for future reference.